src/imgui/ImGuiSymbols.cc

```diff
@@ -188,16 +268,58 @@ void ImGuiSymbols::paint(MSXMotherBoard* /*motherBoard*/)
 		}
 
 		im::TreeNode("Symbols per file", ImGuiTreeNodeFlags_DefaultOpen, [&]{
-			auto drawFile = [&](const FileInfo& info) {
+			auto drawFile = [&](FileInfo& info) {
+				auto fileIdx = symbolManager.findFile(info.filename);
 				im::StyleColor(!info.error.empty(), ImGuiCol_Text, getColor(imColor::ERROR), [&]{
 					auto title = strCat("File: ", info.filename);
 					im::TreeNode(title.c_str(), [&]{
 						if (!info.error.empty()) {
 							ImGui::TextUnformatted(info.error);
 						}
 						im::StyleColor(ImGuiCol_Text, getColor(imColor::TEXT), [&]{
+							auto& cpuInterface = motherBoard->getCPUInterface();
+							std::string slotInfo = {'-', '\000'};
+							std::vector<std::pair<std::optional<uint16_t>, std::optional<uint16_t>>> slotSubslots;
+							slotSubslots.push_back({{}, {}}); // initial state
+							int nSlots = 1;
+
+							for (uint16_t ps = 0; ps < 4; ++ps) {
+								if (cpuInterface.isExpanded(ps)) {
+									slotInfo = tmpStrCat(slotInfo, ps, "-0", '\000', ps, "-1", '\000', ps, "-2", '\000', ps, "-3", '\000');
+									slotSubslots.push_back({ps, 0});
+									slotSubslots.push_back({ps, 1});
+									slotSubslots.push_back({ps, 2});
+									slotSubslots.push_back({ps, 3});
+									nSlots += 4;
+								} else {
+									slotInfo = tmpStrCat(slotInfo, ps, '\000');
+									slotSubslots.push_back({ps, {}});
+									nSlots++;
+								}
+							}
+							auto arrowSize = ImGui::GetFrameHeight();
+							auto extra = arrowSize + 2.0f * style.FramePadding.x;
+							ImGui::SetNextItemWidth(ImGui::CalcTextSize("3-3").x + extra);
+						        ImGui::AlignTextToFramePadding();
+
+							// slot
+							ImGui::Combo(tmpStrCat("Slot##", info.filename).data(), &info.index, slotInfo.c_str(), nSlots);
```

Comment:
You're using `ImGui::Combo()` and because of that you have to construct the `slotInfo` string and store `info.index`. This `ImGui::Combo()` function is a good idea if the list of items in the combo-box is fixed. But for a more dynamic list it's not recommended.

Instead you can use `ImGui::BeginCombo()`, `ImGui::EndCombo()` (or better our `im::Combo()` wrapper around it) combined with `ImGui::Selectable()`. Search the code for some examples.

The basic idea is: with `im::Combo()` you start a combo box context. Inside that context you have your loop of primary and optionally secondary slot. For each existing slot you make a call to `ImGui::Selectable()` (depending in `info.slot` you pass the correct value for the `selected` parameter to `Selectable()`. `Selectable()` also returns a `bool` result. If `true` you update the `info.slot` field to the current slot (i mean the slot in your for loop), and also only then you need to update the slot field in the symbols.